internal/debug/coveruploader/cover_cover.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+//go:build cover
+
+package coveruploader
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"runtime/coverage"
+	"time"
+	_ "unsafe"
+)
+
+// putURL is injected by build scripts as the target for the
+// uploaded coverage profile.
+var putURL string
+
+func Enable() {
+	if putURL == "" {
+		return
+	}
+	ticker := time.NewTicker(10 * time.Second)
+	f := func() {
+		ticker.Stop()
+		upload()
+	}
+	go func() {
+		for {
+			_, ok := <-ticker.C
+			if !ok {
+				return
+			}
+			upload()
+		}
+	}()
+	addExitHook(hook{
+		F:            f,
+		RunOnFailure: true,
+	})
+}
+
+func upload() {
+	cm := &bytes.Buffer{}
+	err := coverage.WriteMeta(cm)
+	if err != nil {
+		log.Printf("cover: cannot write coverage meta: %v\n", err)
```

Comment:
For machines, will this be dumped into logsink? I wonder if just using the internal logger would be better?